common/tree_node.go

```diff
@@ -0,0 +1,247 @@
+package core
+
+import (
+	"fmt"
```

Comment:
At the end, when everything’s working and ready we shouldn’t need fmt since we shouldn’t need to log outside testing.

Reply:
Correct